### PR TITLE
KB-122-Intellect-return-500-error-when-public-profile-not-found

### PR DESCRIPTION
**Pull request by Markusplay**

### CODE REVIEW

File: src/pages/profile/[teacherId].tsx [hunk 1]
```diff
@@ -25,19 +26,27 @@ import { getExperienceByTeacherId, getRatings, getTeacherByTeacherId } from '@/a
...
+            getTeacherByTeacherId(teacherId),
+            getExperienceByTeacherId(teacherId),
+            getRatings(teacherId),
+        ]);
+
+        return {
+            props: { teacher, experience, ratings },
+        };
+    } catch (error) {
+        if (error instanceof AxiosError && error.status === 404) {
```
Comment (a-gubskiy) [inline src/pages/profile/[teacherId].tsx:44]: What should be happening when status is other than 404?

Comment (Markusplay) [reply src/pages/profile/[teacherId].tsx]: adjusted

**Comment by a-gubskiy** [comment]
Screenshot 2024-10-23 at 14 08 51

Seems error handled not as expected.

**Comment by Markusplay** [comment]
> Screenshot 2024-10-23 at 14 08 51
> Seems error handled not as expected.

image
On my local machine look like it works properly.

**Comment by niravzi** [comment]
Same for me:
Screenshot 2024-10-23 at 19 05 07

`status` in AxiosError is `undefined` in this case.

Maybe also create a custom 404 component for this page? Native one looks ugly, breaks layout and styling on a page.

**Comment by Markusplay** [comment]
> Maybe also create a custom 404 component for this page? Native one looks ugly, breaks layout and styling on a page.

Ok, I will handle it if the `status` is `undefined`. 
As we don't have a custom error page in Figma for intellect, I propose to make this page white so it looks native and doesn't break the header. 
Check this out:
image

**Review by a-gubskiy — APPROVED** [review]

**Comment by niravzi** [comment]
Not perfect I would say, but lets not spend more time on this.

**Pull request merged**